extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java

```diff
@@ -220,15 +225,24 @@ void processAnnotationsAndIndexFiles(
             if (xmlSchemaInstance.target().kind() == Kind.CLASS) {
                 String className = xmlSchemaInstance.target().asClass().name().toString();
 
-                reflectiveClass.produce(ReflectiveClassBuildItem.builder(className).build());
+                Type jandexType = Type.create(DotName.createSimple(className), Type.Kind.CLASS);
+                reflectiveClass.produce(new ReflectiveHierarchyBuildItem.Builder()
+                        .type(jandexType)
+                        .index(index)
+                        .source(getClass().getSimpleName() + " > " + jandexType.name().toString())
+                        .build());
             }
         }
 
         // Register XML Java type adapters for reflection
         for (AnnotationInstance xmlJavaTypeAdapterInstance : index.getAnnotations(XML_JAVA_TYPE_ADAPTER)) {
-            reflectiveClass.produce(
-                    ReflectiveClassBuildItem.builder(xmlJavaTypeAdapterInstance.value().asClass().name().toString())
-                            .methods().fields().build());
+            String className = xmlJavaTypeAdapterInstance.value().asClass().name().toString();
+            Type jandexType = Type.create(DotName.createSimple(className), Type.Kind.CLASS);
+            reflectiveClass.produce(new ReflectiveHierarchyBuildItem.Builder()
+                    .type(jandexType)
+                    .index(index)
+                    .source(getClass().getSimpleName() + " > " + jandexType.name().toString())
+                    .build());
```

Comment:
My guess is that this change should be reverted and that's probably what causes the warnings.